### ISSUE TITLE
[`Pylint`] `PLE2510` fix changes behavior with preceding slash

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#18815]
### Summary

If the backspace is preceded by a slash, the fix for [invalid-character-backspace (PLE2510)](https://docs.astral.sh/ruff/rules/invalid-character-backspace/#invalid-character-backspace-ple2510) can change behavior. [playground](https://play.ruff.rs/c6ade0a3-d73f-4319-be04-81cda165e412)
```
PS D:\python_projects> Get-Content issue.py
```
```py
x = "\␈"
print(repr(x))
```
^ note: I'm cheating here, PowerShell does not actually display the backspace character as `␈`, it actually gets displayed as `x = ""`
```
PS D:\python_projects> py issue.py
```
```
D:\python_projects\issue.py:1: SyntaxWarning: invalid escape sequence ''
  x = ""
'\\\x08'
```
```
PS D:\python_projects> uvx ruff check issue.py --isolated --select PLE
```
```snap
issue.py:1:7: PLE2510 [*] Invalid unescaped character backspace, use "\b" instead
  |
1 | x = "\␈"
  |       ^ PLE2510
2 | print(repr(x))
  |
  = help: Replace with escape sequence

Found 1 error.
[*] 1 fixable with the `--fix` option.
```
```
PS D:\python_projects> uvx ruff check issue.py --isolated --select PLE --fix
```
```
Found 1 error (1 fixed, 0 remaining).
```
```
PS D:\python_projects> Get-Content issue.py
```
```py
x = "\\b"
print(repr(x))
```
```
PS D:\python_projects> py issue.py
```
```
'\\b'
```

This also applies to `PLE2512`, `PLE2513`, and `PLE2515` [playground](https://play.ruff.rs/dc2bc095-7f43-4f4c-ba03-a1167e147196), as well as `PLE2514`, but the playground does not support null chars.

### Version

ruff 0.12.0 (87f0feb 2025-06-17) + playground

**Comment by MichaReiser** [comment]
I think what we should do here is to check if there's an uneven number of preceding `\` and if so, not provide a fix.
